tool/src/org/antlr/v4/tool/ErrorType.java

```diff
@@ -955,6 +955,12 @@
 	 * <p>custom channels are not supported in combined grammars</p>
 	 */
 	CHANNELS_BLOCK_IN_COMBINED_GRAMMAR(164, "custom channels are not supported in combined grammars", ErrorSeverity.ERROR),
+	/**
+	 * Compiler Error 165.
+	 *
+	 * <p>mode <em>name</em> conflicts with token with same name</p>
+	 */
+	MODE_CONFLICTS_WITH_TOKEN(165, "mode <arg> conflicts with token with same name", ErrorSeverity.ERROR),
```

Comment:
:exclamation: Errors 165-168 are reserved for #697. They may never end up used, but that is OK since we still have the advantage of avoiding a conflict with my optimized fork (which already includes a preview implementation of that feature).

:bulb: Consider expanding the documentation for this rule. See compiler warning 147 for an example of more complete documentation for a warning or error.